src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,10 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user)
+                .filter(eachUser -> eachUser.getAge() != null)
+                .filter(eachUser -> eachUser.getAge() >= 30 && eachUser.getAge() <= 45)
```

Comment:
getAge가 여러번 호출되고 있어요, map으로 getAge를 먼저 해주면 어떨까요?

```suggestion
                .map(User::getAge)
```